src/blocks/libvirt.rs

```diff
@@ -24,6 +24,25 @@ pub struct Libvirt {
     qemu_conn: Connect,
 }
 
+impl Libvirt {
+    fn check_and_reconnect(&mut self) -> Result<&mut Self> {
+        if !self
+            .qemu_conn
+            .is_alive()
+            .block_error("virt", "error when getting connection status to libvirt")?
+        {
+            self.qemu_conn = Connect::open_read_only(
+                &self
+                    .qemu_conn
+                    .get_uri()
+                    .block_error("virt", "error in retrieving URI information from libvirt")?,
+            )
+            .block_error("virt", "error in reconnecting to libvirt socket")?;
+        };
+        Ok(self)
```

Comment:
I'm not sure why you need to return `&mut Self` here.

Reply:
I think the preview is not linking to the right line anymore; but assuming you mean the `impl Libvirt {}` block: it's because I modify the struct parameter `qemu_conn` if I reconnect the endpoint.

Reply:
I meant that you can return `Result<()>` instead of `Result<&mut Self>` from `check_and_reconnect()`.

Reply:
AH! OK. Yes, that's good feedback. Thank you!